backend/uclapi/roombookings/views.py

```diff
@@ -163,3 +164,66 @@ def get_equipment(request, *args, **kwargs):
         "ok": True,
         "equipment": _serialize_equipment(equipment)
     }, rate_limiting_data=kwargs)
+
+
+@api_view(['GET'])
+# @uclapi_protected_endpoint()
+def get_free_rooms(request, *args, **kwargs):
+    request_params = {}
+    request_params['startdatetime__gte'] = request.GET.get('start_datetime')
+    request_params['finishdatetime__lte'] = request.GET.get('end_datetime')
+
+    if (
+        not request_params['startdatetime__gte'] or
+        not request_params['finishdatetime__lte']
+    ):
+        return PrettyJsonResponse({
+            "ok": False,
+            "error": "start_datetime or end_datetime not provided"
+        }, rate_limiting_data=kwargs)
+
+    is_parsed = True
+
+    start, end, is_parsed = _parse_datetime(
+        request_params['startdatetime__gte'],
+        request_params['finishdatetime__lte'],
+        None
+    )
+
+    if not is_parsed:
+        return PrettyJsonResponse({
+            "ok": False,
+            "error": "date/time isn't formatted as suggested in the docs"
+        }, rate_limiting_data=kwargs)
+
+    # Rounding down start date to start of day
+    request_params["startdatetime__gte"] = _round_date(start)
+
+    # Rounding up end date to start of next day
+    request_params["finishdatetime__lte"] = _round_date(end, up=True)
+
+    # Pagination Logic
+    # maxing out results_per_page to get all the bookings in one page
+    results_per_page = 100000
```

Comment:
@anirudhpillai I don't fully understand this line. Is the idea just that we skip pagination entirely?

Reply:
Yeah, we're trying to get everything in one page.

Reply:
Yes, that's what it is doing. This lets us reuse the caching function written by Faiz. Also, it shouldn't actually return those many bookings as we only look up bookings between the given start and end date.